drivers/bus/isapnp/isapnp.c

```diff
@@ -829,51 +1476,52 @@ IsaPower(
     return PoCallDriver(((PISAPNP_FDO_EXTENSION)DevExt)->Ldo, Irp);
 }
 
-static DRIVER_DISPATCH IsaPnp;
+_Dispatch_type_(IRP_MJ_PNP)
+static DRIVER_DISPATCH_PAGED IsaPnp;
 
 static
+CODE_SEG("PAGE")
 NTSTATUS
 NTAPI
 IsaPnp(
-    IN PDEVICE_OBJECT DeviceObject,
-    IN PIRP Irp)
+    _In_ PDEVICE_OBJECT DeviceObject,
+    _Inout_ PIRP Irp)
 {
     PIO_STACK_LOCATION IrpSp = IoGetCurrentIrpStackLocation(Irp);
     PISAPNP_COMMON_EXTENSION DevExt = DeviceObject->DeviceExtension;
 
-    DPRINT("%s(%p, %p)\n", __FUNCTION__, DeviceObject, Irp);
+    PAGED_CODE();
 
     if (DevExt->IsFdo)
-    {
-       return IsaFdoPnp((PISAPNP_FDO_EXTENSION)DevExt,
-                        Irp,
-                        IrpSp);
-    }
+        return IsaFdoPnp((PISAPNP_FDO_EXTENSION)DevExt, Irp, IrpSp);
     else
-    {
-       return IsaPdoPnp((PISAPNP_PDO_EXTENSION)DevExt,
-                        Irp,
-                        IrpSp);
-    }
+        return IsaPdoPnp((PISAPNP_PDO_EXTENSION)DevExt, Irp, IrpSp);
 }
 
+CODE_SEG("INIT")
 NTSTATUS
 NTAPI
 DriverEntry(
-    IN PDRIVER_OBJECT DriverObject,
-    IN PUNICODE_STRING RegistryPath)
+    _In_ PDRIVER_OBJECT DriverObject,
+    _In_ PUNICODE_STRING RegistryPath)
 {
     DPRINT("%s(%p, %wZ)\n", __FUNCTION__, DriverObject, RegistryPath);
 
     DriverObject->MajorFunction[IRP_MJ_CREATE] = IsaCreateClose;
     DriverObject->MajorFunction[IRP_MJ_CLOSE] = IsaCreateClose;
-    DriverObject->MajorFunction[IRP_MJ_READ] = IsaReadWrite;
-    DriverObject->MajorFunction[IRP_MJ_WRITE] = IsaReadWrite;
-    DriverObject->MajorFunction[IRP_MJ_DEVICE_CONTROL] = IsaIoctl;
+    DriverObject->MajorFunction[IRP_MJ_DEVICE_CONTROL] = IsaForwardOrIgnore;
+    DriverObject->MajorFunction[IRP_MJ_SYSTEM_CONTROL] = IsaForwardOrIgnore;
     DriverObject->MajorFunction[IRP_MJ_PNP] = IsaPnp;
     DriverObject->MajorFunction[IRP_MJ_POWER] = IsaPower;
     DriverObject->DriverExtension->AddDevice = IsaAddDevice;
 
+//  _No_competing_thread_begin_
+
+    KeInitializeEvent(&BusSyncEvent, SynchronizationEvent, TRUE);
+    InitializeListHead(&BusListHead);
+
+//  _No_competing_thread_end_
```

Comment:
What are these comments for?

Reply:
This annotation will be enabled later, SDK needs investigation

Reply:
yeah, I've checked our headers and actually this annotation is present in the sal headers (at least the "no_sal" one)